elastic/kibana
# [Actions] Gracefully handle timestamp fields that are empty or have spaces in PagerDuty action params

Resolves https://github.com/elastic/kibana/issues/100042

## Summary

Updated the PagerDuty connector type to be more resilient to empty strings or strings with spaces in the `timestamp` field.
* If the timestamp field is set to an empty string, it will be removed from the payload sent to PagerDuty
* Trimmed the timestamp field so extraneous spaces are removed.

## To Verify
* Create a rule, either through the UI or API with a PagerDuty action, where the `timestamp` field is set to `''`. Verify the PagerDuty action runs successfully without erroring and the PagerDuty event contains no timestamp.
* Create a rule, either through the UI or API with a PagerDuty action, where the `timestamp` field is set to a context variable that doesn't exist (`{{context.fakeTimestamp}}` for example). Verify the PagerDuty action runs successfully without erroring and the PagerDuty event contains no timestamp.
* Create a rule, either through the UI or API with a PagerDuty action, where the `timestamp` field is set to a valid date time but contains extraneous space (`  {{date}}  ` should work). Verify the PagerDuty action runs successfully without erroring and the PagerDuty event does contains a timestamp.

### Checklist

Delete any items that are not applicable to this PR.
